File: algorithms/cpp/376.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int wiggleMaxLength(vector<int> &nums) {
        if (nums.size() == 0)
            return 0;
        int a = 1, b = 1;
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] > nums[i-1])
                a = b+1;
            if (nums[i] < nums[i-1])
                b = a+1;
        }
        return max(a, b);
    }
};

int main() {
    return 0;
}

File: algorithms/python/376.py
class Solution:
    def wiggleMaxLength(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        l = len(nums)
        if l <= 1:
            return l
        state, rls = 0, 1
        for i in range(1, l):
            diff = nums[i] - nums[i-1]
            if state >= 0 and diff < 0:
                state = diff
                rls += 1
            elif state <= 0 and diff > 0:
                state = diff
                rls += 1
        return rls
